### DISCUSSION TITLE
Using multiples instances of aircrafts/engines in Unreal Engine

### DISCUSSION THREAD

**Discussion by Franceis** [JSBSim-Team/jsbsim#966] [Q&A]
Hello, how are you?

I'm a new user of JSBSim, and I'm conducting some tests using it with Unreal Engine. In my project, I'm experimenting with using multiple instances of aircraft simultaneously in a single simulation, all controlled by a Unreal class called "AIController," which allows them to be piloted through AI without requiring user input for the controlled aircraft.

However, while reading the latest plugin notes, I came across a sentence stating that "Multiple instances of JSBSim components/aircrafts currently do not work. You can only run one aircraft simulation at a time." This raised the question of what exactly happens or could happen if I were to disregard this note and attempt to use multiple instances of aircraft/engines in the same simulation in my project. The potential obstacles that I might encounter while trying to work with this system are not very well described. Could you help me specify them?

Thank you in advance.

**Comment by bcoconni** [comment]
@gallonmate, @demonixis, @AlbanBERGERET-Epic any reason about why multiple instances of JSBSim cannot be used with the UE plugin at the moment ?
And maybe some tips about what it would require to allow the UE plugin using multiple instances of JSBSim ?

Thanks.

**Comment by demonixis** [comment]
I'm not sure but I'm pretty sure I was able to use two airplanes at the same time. The code doesn't use static stuff, but use a lot of std::shared_ptr<T>, maybe we can replace them with another smart ptr which is unique to the class.

**Comment by AlbanBERGERET-Epic** [comment]
I haven't really tried it... 

The component initializes everything. If there are some initialisation that could be shared among several components, a World SubSystem would make sense. But as of today, since the computations are not done in a separate thread, we would probably hit some performance issues with some aircrafts at the same time.
